src/components/ConfigEditor/ConfigEditor.jsx

```diff
@@ -0,0 +1,178 @@
+import React, { useState, useEffect } from "react";
+import useSWR from "swr";
+import { getHintOptions } from "vitessce";
+
+import { StudyIdInput } from "../StudyIdInput";
+import { baseJson, exampleConfig } from "../../utils/config-examples.js";
+import { ErrorDiv } from "../ErrorDiv";
+
+import styles from "./ConfigEditor.module.css";
+
+import {
+	ERROR_MESSAGES,
+	EXAMPLE_URL,
+	LINK_ID_ENDPOINT_URL,
+	LINK_ID_LENGTH,
+	LINK_ID_KEY,
+} from "../../utils/constants.js";
+import {
+	validateConfig,
+	sanitizeURLs,
+	studyIdFetcher,
+	updateConfigWithExampleURL,
+} from "../../utils/utility-functions.js";
+
+export const ConfigEditor = ({
+	pendingJson,
+	setPendingJson,
+	error,
+	setError,
+	setLoading,
+	setUrl,
+	setLinkIdInput,
+}) => {
+	const [datasetUrls, setDatasetUrls] = useState("");
+	const [generateConfigError, setGenerateConfigError] = useState(null);
+	const [studyId, setStudyId] = useState(null);
+	const [linkId, setLinkId] = useState(null);
+	const [linkIdUrl, setLinkIdUrl] = useState(null);
+	const [showReset, setShowReset] = useState(null);
```

Comment:
Similar thought with `showReset` - it gets set to `true` when `handleConfigGeneration` is invoked (i.e. when the user starts editing `datasetUrls`, and gets reset to false by `resetEditor`, which clears `datasetUrls`/sets it back to an empty string.

Perhaps `showReset` could be calculated based on the datasetUrls state being non-empty instead to trim down complexity? e.g.

```tsx
const showReset = Boolean(datasetUrls)
```